backend/packages/Upgrade/src/api/controllers/StratificationController.ts

```diff
@@ -0,0 +1,240 @@
+   *      produces:
+   *        - application/json
+   *      parameters:
+   *        - in: body
+   *          name: file
+   *          description: CSV file
+   *          required: true
+   *          schema:
+   *            type: file
+   *      responses:
+   *        '200':
+   *          description: Create a new UserStratificationFactors by CSV
+   *          schema:
+   *            type: array
+   *            items:
+   *              $ref: '#/definitions/UserStratificationFactor'
+   *        '401':
+   *          description: Authorization Required Error
+   *        '500':
+   *          description: Internal Server Error, Insert Error in database, CSV file is not valid
+   */
+  @Post()
+  @UseBefore(upload.single('file'))
+  public insertStratification(@Req() request: AppRequest): Promise<UserStratificationFactor[]> {
+    const csvData = request.body[0].file;
+    const rows = csvData.replace(/"/g, '').split('\n');
+    const columnNames = rows[0].split(',');
+
+    const userFactorValues: StratificationInputValidator[] = [];
+
+    // Iterate through the rows (skip the header row)
+    for (let i = 1; i < rows.length; i++) {
+      const row = rows[i];
+      const rowValues = row.split(',');
+
+      // Extract the user ID
+      const userId = rowValues[0];
+      if (!userId) {
+        continue;
+      }
+
+      // Iterate through other columns (factors)
+      for (let j = 1; j < columnNames.length; j++) {
+        const factorName = columnNames[j];
+        const factorValue = rowValues[j].trim();
+
+        // Create an object and add it to the array
+        const userFactorValue: StratificationInputValidator = {
+          userId: userId,
+          factor: factorName.trim(),
+          value: factorValue === '' ? null : factorValue,
+        };
+
+        userFactorValues.push(userFactorValue);
+      }
+    }
+
+    return this.stratificatonService.insertStratification(userFactorValues, request.logger);
+  }
+
+  /**
+   * @swagger
+   * /stratification/{factor}:
+   *    delete:
+   *      description: Delete a stratificationFactor by factorName
+   *      tags:
+   *      - Stratification
+   *      produces:
+   *        - application/json
+   *      parameters:
+   *        - in: path
+   *          name: factor
+   *          description: Factor name
+   *          required: true
+   *          schema:
+   *            type: string
+   *      responses:
+   *        '200':
+   *          description: Delete a stratificationFactor by factorName
+   *          schema:
+   *            $ref: '#/definitions/FactorStrata'
+   *        '401':
+   *          description: Authorization Required Error
+   *        '500':
+   *          description: Internal Server Error, FactorName is not valid
+   */
+  @Delete('/:factor')
+  public deleteStratification(
+    @Param('factor') stratificationFactor: string,
+    @Req() request: AppRequest
+  ): Promise<StratificationFactor> {
+    if (!stratificationFactor) {
```

Comment:
Can you add class-validator decorator with the @Param('factor') for not empty?

Reply:
can't find the right syntax

Reply:
Can you try @Params decorator it has a validation option. The only drawback is that you will get an object of all the params.

Reply:
is this resolved?